lrlex/src/lib/parser.rs

```diff
@@ -221,11 +224,144 @@ where
         }
     }
 
+    fn parse_grmtools_section_value(
+        &mut self,
+        mut i: usize,
+        span_map: &mut HashMap<&str, Span>,
+    ) -> LexInternalBuildResult<usize> {
+        const OPTIONS: [&str; 11] = [
+            "dot_matches_new_line",
+            "multi_line",
+            "octal",
+            "posix_escapes",
+            "case_insensitive",
+            "swap_greed",
+            "ignore_whitespace",
+            "unicode",
+            "size_limit",
+            "dfa_size_limit",
+            "nest_limit",
+        ];
+        let start_pos = i;
+        // RegexBuilder isn't uniform regarding whether the default value of an options is true
+        // or false. For instance `unicode` is `true` but `case_insensitive` defaults to false.
+        let flag = if self.src[i..].starts_with("!") {
+            i += 1;
+            false
+        } else {
+            true
+        };
+        for opt in OPTIONS {
+            if self.src[i..].to_lowercase().starts_with(opt) {
+                match opt {
+                    "dot_matches_new_line" | "multi_line" | "octal" | "posix_escapes" => {
+                        if let Some(orig_span) = span_map.get(opt) {
+                            return Err(LexBuildError {
+                                kind: LexErrorKind::DuplicateGrmtoolsSectionValue,
+                                spans: vec![*orig_span, Span::new(start_pos, i)],
+                            });
+                        }
+                        match opt {
+                            "dot_matches_new_line" => {
+                                self.regex_options.dot_matches_new_line = flag
+                            }
+                            "multi_line" => self.regex_options.multi_line = flag,
+                            "octal" => self.regex_options.octal = flag,
+                            "posix_escapes" => self.regex_options.posix_escapes = flag,
+                            _ => unreachable!(),
+                        }
+                        i += opt.len();
+                        span_map.insert(opt, Span::new(start_pos, i));
+                        return self.parse_nl(i);
```

Comment:
Is the `return` here intentional? It surprised me a bit.

Reply:
Not really, does seem like we should just let fall though to the `parse_ws` call below doesn't it.

Reply:
I *think* so. Indeed, I kind of expected all of these clauses to end in the same way, so the differences did jump out at me a bit.

Reply:
Should be fixed in [509296c](https://github.com/softdevteam/grmtools/pull/496/commits/509296c9958edc3bc2cef2d29a61aa5b049df528)